collector/cpu_linux.go

```diff
@@ -68,6 +71,17 @@ func NewCPUCollector(logger log.Logger) (Collector, error) {
 	if err != nil {
 		return nil, fmt.Errorf("failed to open procfs: %w", err)
 	}
+
+	sysfs, err := sysfs.NewFS(*sysPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to open sysfs: %w", err)
+	}
+
+	isolcpus, err := sysfs.IsolatedCPUs()
```

Comment:
Since which kernel version does that file exist? Maybe we should just skip the metric if it doesn't exist.

Reply:
I couldn't find a definitive answer after grepping the kernel source for a while.
It's definitely there in 3.10, and not there in 2.6.31. I don't have anything running in-between
I've defaulted now to an empty slice if there's any error

Reply:
I couldn't figure out how to run another e2e test with a missing file. Am I missing something obvious or is this not possible ?

Reply:
Since 4.2 actually! [commit](https://github.com/torvalds/linux/commit/59f30abe94bff50636c8cad45207a01fdcb2ee49) but redhat's 3.10 has this backported

Reply:
Okay, I guess though it won't harm to just skip exposing the metric when it's not available.

So I'd check if the error is unix.ENOENT, log a warning with level Debug, then check if isolcpus != nil here: https://github.com/prometheus/node_exporter/pull/2251/files#diff-7f89527fd186b6ae37f9e36c257f2c48bd15f09596ceaf488d3325d470d4933fR166

Reply:
I ended up checking for `os.IsNotExist` as `err` was always != from `unix.ENOENT`; should i change something in `procfs`?

Reply:
No that's fine! Thanks!